src/huggingface_hub/keras_mixin.py

```diff
@@ -15,6 +19,119 @@
 
 
 logger = logging.getLogger(__name__)
+if is_tf_available():
+    import tensorflow as tf
+
+
+def _extract_hyperparameters_from_keras(model):
+    hyperparameters = dict()
+    if hasattr(model, "optimizer") and model.optimizer is not None:
```

Comment:
If it's `None`, you still want to assign it, so this should work

```suggestion
    if hasattr(model, "optimizer"):
```